lms/djangoapps/certificates/tests/test_support_views.py

```diff
@@ -326,6 +331,33 @@ def test_regenerate_user_has_no_certificate(self):
         num_certs = GeneratedCertificate.eligible_certificates.filter(user=self.student).count()
         self.assertEqual(num_certs, 1)
 
+    def test_regenerate_user_with_invalidated_cert(self):
+        """ Verify if the certificate is marked as invalid, if so, regenerate
```

Comment:
`If the certificate is marked as invalid, regenerate the certificate and verify the invalidate entry is deactivated`